sssd_test_framework/hosts/base.py

```diff
@@ -296,3 +296,69 @@ def __init__(self, *args, **kwargs) -> None:
 
         self.fs: LinuxFileSystem = LinuxFileSystem(self)
         self.svc: SystemdServices = SystemdServices(self)
+        self._os_release: dict = {}
+        self._distro_name: str | None = None
+        self._distro_major: int | None = None
+        self._distro_minor: int | None = None
+
+    def _distro_information(self):
+        """
+        Pulls distro information from a host from /ets/os-release
+        """
+        self.logger.info(f"Detecting distro information on {self.hostname}")
+        os_release = self.ssh.run("cat /etc/os-release", log_level=SSHLog.Error).stdout_lines
+        for line in os_release:
+            if "=" not in line:
+                continue
+            k, v = line.rstrip().split("=", maxsplit=1)
+            self._os_release[k] = v.strip('"')
+        if "NAME" in self._os_release:
+            self._distro_name = self._os_release["NAME"]
+        if "VERSION_ID" not in self._os_release:
+            return
+        if "." in self._os_release["VERSION_ID"]:
+            self._distro_major = int(self._os_release["VERSION_ID"].split(".", maxsplit=1)[0])
+            self._distro_minor = int(self._os_release["VERSION_ID"].split(".", maxsplit=1)[1])
+        else:
+            self._distro_major = int(self._os_release["VERSION_ID"])
+            self._distro_minor = 0
+
+    @property
+    def distro_name(self) -> str | None:
```

Comment:
Do you really want to return None as a possible value? I think it would be better to keep them always non-None with default value (name=unknown, version=0.0), otherwise you will have to assert that it is not None before every use.

Reply:
Ack. Initially I wanted to use None when I did not try to detect distro yet but I can use self._os_release for that.